annotate.py

```diff
@@ -212,20 +214,19 @@ def annotate_img(img_path, initials, size1=1, size2=1) :
 
     #Make mask same colour as drawing and output binarised image
     train_labels = img[:,:,:3]
-    lower_blue = np.array([254,0,0], dtype = "uint16")
-    upper_blue = np.array([255,0,0], dtype = "uint16")
-    mask_blue = cv2.inRange(train_labels, lower_blue, upper_blue)
-    mask_blue[mask_blue < 250] = 0
-    #mask_blue[mask_blue != 0 ] = 255
-    mask_blue[mask_blue != 0 ] = 1
+    lower_red = np.array(RED_MIN, dtype = "uint16")
+    upper_red = np.array(RED_MAX, dtype = "uint16")
+    mask_red = cv2.inRange(train_labels, lower_red, upper_red)
+    mask_red[mask_red < COLOR_THRESHOLD] = 0
+    mask_red[mask_red != 0 ] = 1
 
 
-    lower_red = np.array([0,0,254], dtype = "uint16")
-    upper_red = np.array([0,0,255], dtype = "uint16")
-    mask_red = cv2.inRange(train_labels, lower_red, upper_red)
-    mask_red[mask_red < 250] = 0
+    lower_blue = np.array(BLUE_MIN, dtype = "uint16")
```

Comment:
From here, this is actually more copy/pasted code from the `RED` section. Considering the above comments, in principle the only input you really need is `255`, which can then be used to define the rest of the variables, including both colours.

Consider using a list or dictionary that you can loop through, using only a single block of code. This will also help for generality, in case later we want to add a third feature in the `green` channel (or beyond with any number of colours, but will need to update the `np` tricks above).

Reply:
Sounds good, I'll change the code to have it loop through a dictionary of colors, and only keep one `255` variable that affects the rest.